src/ast.rs

```diff
@@ -301,15 +308,20 @@ impl<'i> Parser<'i> {
           let nam = self.parse_name()?;
           Ok(Tree::Ref { nam })
         }
-        // Num = "#" Int
+        // Int = "#" Int
+        // F32 = "#" Int "." Int
         Some('#') => {
           self.advance_char();
-          match self.peek_char() {
-            Some('-') => {
-              self.advance_char();
-              Ok(Tree::Num { val: -(self.parse_int()? as i64) })
-            }
-            _ => Ok(Tree::Num { val: self.parse_int()? as i64 }),
+          let is_neg = self.consume("-").is_ok();
+          let int_val = if is_neg { -(self.parse_int()? as i64) } else { self.parse_int()? as i64 };
+
+          if self.consume(".").is_ok() {
+            let frac_val = self.parse_int()?;
+            let frac_len = frac_val.checked_ilog10().unwrap_or(0) + 1;
```

Comment:
```rs
$ cargo run reduce -- '#1.0001'
#1.1
```

IMO a slight amount of backtracking here is fine

Also, there should be syntax for NaN and +/- infinity